client/landing/jetpack-cloud/components/date-picker/index.jsx

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { withLocalizedMoment } from 'components/localized-moment';
+import Button from 'components/forms/form-button';
+import DateRangeSelector from 'my-sites/activity/filterbar/date-range-selector';
+import Gridicon from 'components/gridicon';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+class DatePicker extends Component {
+	state = {
+		currentSetting: false,
+	};
+
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		initialDate: PropTypes.instanceOf( Date ).isRequired,
+		onChange: PropTypes.func.isRequired,
+	};
+
+	getFormattedDate = date => this.props.moment( date ).format( DATE_FORMAT );
+
+	getDisplayDate = date => {
+		const word = this.props
+			.moment( date )
+			.calendar()
+			.split( ' ' )[ 0 ];
+		if ( 'Today' === word || 'Yesterday' === word ) {
+			return word;
+		}
+
+		return this.getFormattedDate( date );
+	};
+
+	shuttleLeft = () => {
+		const { initialDate, moment } = this.props;
+		const currentSetting = moment(
+			this.state.currentSetting ? this.state.currentSetting : initialDate
+		).subtract( 1, 'days' );
+		this.setState( { currentSetting } );
+		this.props.onChange( currentSetting.format( DATE_FORMAT ) );
+	};
+
+	shuttleRight = () => {
```

Comment:
Same comment as above

Reply:
Sure, same as shuttleLeft, this function is connected to the right shuttle button, which moves the date into the future if possible.